mvcc/backend/backend.go

```diff
@@ -119,6 +120,13 @@ func NewDefaultBackend(path string) Backend {
 }
 
 func newBackend(bcfg BackendConfig) *backend {
+	// snapshotWarningTimeout is based on transferring 4 GB over a 1 gigabit/s connection
+	// with max tcp throughput of 949.285 Mbit/s limited by TCP overhead.
+	// every 4 gigabytes increases snapshotWarningTimeout by 35 seconds.
+	// e.g size(db) <= 4 GB takes < 35 seconds, size(db) <= 8 GB takes < 70 seconds.
+	factor := bcfg.MmapSize/(4*1024*1024*1024) + 1
```

Comment:
probably have a min warning duration: 30seconds. if we know the db size is large, we increase 30 seconds to something more reasonable.